User.java

```diff
@@ -0,0 +1,45 @@
+import java.util.HashMap;
+
+public class User {
+    int id;
+    String name;
+    double balance;
+    HashMap<Thing, Integer> things;
```

Comment:
Can be named thingsMap, even though things is a bad name as discussed above, this can also be improved by naming the variable with the specific implementation. When we see the variable being used later on, it will be very obvious we are dealing with a map type object if the var is called, "thingsMap", without us having to check the type.